spec/scenarios/terminate_pay_in_advance_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Terminate Pay in Advance Scenarios', :scenarios, type: :request do
+  let(:organization) { create(:organization, webhook_url: nil) }
+  let(:customer) { create(:customer, organization:) }
+  let(:plan) { create(:plan, pay_in_advance: true, organization:, amount_cents: 5000) }
+  let(:metric) { create(:billable_metric, organization:) }
+
+  it 'creates expected credit note and invoice' do
+    ### 8 Feb: Create and terminate subscription
+    feb8 = DateTime.new(2023, 2, 8)
+
+    travel_to(feb8) do
+      expect {
+        create_subscription(
+          {
+            external_customer_id: customer.external_id,
+            external_id: customer.external_id,
+            plan_code: plan.code,
+          },
+        )
+      }.to change(Invoice, :count).by(1)
+
+      subscription = customer.subscriptions.find_by(external_id: customer.external_id)
+      sub_invoice = subscription.invoices.first
+      expect(sub_invoice.total_amount_cents).to eq(4500) # 60 / 28 * 21
+
+      expect {
+        terminate_subscription(subscription)
+      }.to change { subscription.reload.status }.from('active').to('terminated')
+        .and change { subscription.invoices.count }.from(1).to(2)
+        .and change { sub_invoice.reload.credit_notes.count }.from(0).to(1)
+
+      term_invoice = subscription.invoices.order(sequential_id: :desc).first
+      expect(term_invoice.total_amount_cents).to eq(0)
+
+      credit_note = sub_invoice.credit_notes.first
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `credit_note`. (https://rubystyle.guide#underscore-unused-vars)